Osiris/Hacks/Misc.cpp

```diff
@@ -335,6 +345,111 @@ void Misc::spectatorList() noexcept
+
+void Misc::damageList() noexcept
+{
+    static int rows_count;
+
+    if (!miscConfig.damageList.enabled)
+        return;
+
+    if ((damageCount.size() == 0 || !interfaces->engine->isInGame() || !interfaces->engine->isConnected()) && !gui->isOpen())
+        return;
+    
+    if (miscConfig.damageList.pos != ImVec2{}) {
+        ImGui::SetNextWindowPos(miscConfig.damageList.pos);
+        miscConfig.damageList.pos = {};
+    }
+
+    if (miscConfig.damageList.size != ImVec2{}) {
+        ImGui::SetNextWindowSize(ImClamp(miscConfig.damageList.size, {}, ImGui::GetIO().DisplaySize));
+        miscConfig.damageList.size = {};
+    }
+
+    ImGuiWindowFlags windowFlags = ImGuiWindowFlags_NoCollapse;
+    if (!gui->isOpen())
+        windowFlags |= ImGuiWindowFlags_NoInputs;
+    if (miscConfig.damageList.noTitleBar)
+        windowFlags |= ImGuiWindowFlags_NoTitleBar;
+
+    if (!gui->isOpen())
+        ImGui::PushStyleColor(ImGuiCol_TitleBg, ImGui::GetColorU32(ImGuiCol_TitleBgActive));
+
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowTitleAlign, { 0.5f, 0.5f });
+    ImGui::Begin("Damage list", nullptr, windowFlags);
+    ImGui::PopStyleVar();
+
+    if (!gui->isOpen())
+        ImGui::PopStyleColor();
+
+    rows_count = 0;
+
+    for (size_t i = 0; i < damageCount.size(); i++) {
+
+        if (rows_count >= miscConfig.damageList.maxRows)
+            break;
+
+        if (damageCount[i].damage == 0)
+            continue;
+
+        if (const auto playerdata = GameData::playerByHandle(damageCount[i].playerHandle)) {
+            if (const auto texture = playerdata->getAvatarTexture()) {
+                const auto text_size = ImGui::CalcTextSize(playerdata->name.c_str());
+                ImGui::Image(texture, ImVec2(text_size.y, text_size.y), ImVec2(0, 0), ImVec2(1, 1), ImVec4(1, 1, 1, 1), ImVec4(1, 1, 1, 0.3f));
+                ImGui::SameLine();
+                ImGui::TextWrapped("%s: ", playerdata->name.c_str());
+                ImGui::SameLine(0.f, 1.f);
+                if(i == 0)
+                    ImGui::PushStyleColor(ImGuiCol_Text, IM_COL32(255, 0, 0, 255));
+                ImGui::TextWrapped("%i", damageCount[i].damage);
+                if (i == 0)
+                    ImGui::PopStyleColor();
+                rows_count++;
+            }
+        }
+    }
+
+    ImGui::End();
+}
+
+void Misc::addDamageToDamageList(GameEvent& event) noexcept
+{
+    const auto attacker = event.getInt("attacker");
+    const auto dmg_health = event.getInt("dmg_health");
+    const auto handle = interfaces->entityList->getEntity(interfaces->engine->getPlayerForUserID(attacker))->handle();
+
+    static bool found;
+    found = false;
+
+    for (auto& target : damageCount) {
+        if (target.playerHandle == handle) {
+            target.damage += dmg_health;
+            found = true;
+            break;
+        }
+    }
+
+    if (!found)
+        damageCount.push_back(DamageData(dmg_health, handle));
```

Comment:
For this kind of usage a `std::unordered_map` would be better. With a map you could do `damageCount[handle] += dmg_health;` instead of those lines of code.

Reply:
`std::unordered_map` cannot be sorted